### PR TITLE
Remove seamless-immutable

### PR DESCRIPTION
**Pull request by tynes**
- Remove `seamless-immutable`
- Add `safeSet` method + tests

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -14,6 +14,9 @@
       "helpers": true,
       "polyfill": true,
       "regenerator": true
+    }],
+    ["transform-object-rest-spread", {
+      "useBuiltIns": true
```
Comment (bucko13) [inline .babelrc]: Will we need to add this (or should we add this) to the bpanel-cli plugin template?

File: lib/index.js [hunk 2]
```diff
@@ -27,6 +27,8 @@ import {
   createAccount
 } from './actions';
 
+import { safeSet } from './utilities';
```
Comment (bucko13) [inline lib/index.js]: Do you see this being useful in other situations? Does it make sense to add to bpanel-utils? or is it too specific? Almost feels like you could publish it as it's own stand alone npm module! :D

File: lib/utilities.js [hunk 3]
```diff
@@ -13,6 +13,26 @@ context ${context}`);
   }
 };
 
+// safely set a value nested deeply in an objects
+export const safeSet = (obj, dotpath, data) => {
```
Comment (bucko13) [inline lib/utilities.js]: This makes a lot more sense to me now looking at the code. Simple and elegant. Very cool!

File: tests/utilities.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+const { assert } = require('chai');
```
Comment (bucko13) [inline tests/utilities.js]: Tests! 🎊 🎈

File: lib/index.js (hunk 2 above)
Comment (tynes) [reply lib/index.js]: Thanks! It could be useful in `bpanel-utils` if we test it with more edge cases

File: lib/utilities.js (hunk 3 above)
Comment (tynes) [reply lib/utilities.js]: Should I add a better comment?

File: .babelrc (hunk 1 above)
Comment (tynes) [reply .babelrc]: It gives a warning when running -mocha --compilers js:babel-core/register tests` that `--compilers` is going to be deprecated, so maybe if we first fix that

File: lib/utilities.js (hunk 3 above)
Comment (bucko13) [reply lib/utilities.js]: I think this is fine. Just wasn't as clear to me when talking about it in words.
